src/data/types.ts

```diff
@@ -1,26 +1,29 @@
 import { ExcalidrawElement } from "../element/types";
 import { AppState, LibraryItems } from "../types";
+import type { cleanAppStateForExport } from "../appState";
 
-export interface DataState {
-  type?: string;
-  version?: string;
-  source?: string;
+export interface ExportedDataState {
+  type: string;
+  version: number;
+  source: string;
   elements: readonly ExcalidrawElement[];
-  appState: Omit<AppState, "offsetTop" | "offsetLeft" | "width" | "height">;
+  appState: ReturnType<typeof cleanAppStateForExport>;
 }
 
 export interface ImportedDataState {
   type?: string;
-  version?: string;
+  version?: number;
   source?: string;
```

Comment:
do we use this anywhere?

Reply:
> This would be a breaking change for hosts so will need to mention it in changelog under refactor

Yeah, but basically any change to any of our internal type is technically a breaking change since we export every type.

Reply:
Not at the moment, but we need to track it lest in the future we need to make non-trivial migrations.

Reply:
hmm ok so this is nowhere mentioned in readme so that is  why I wasn't sure if this is required